Building fails when using -DSYSTEM_YAMLCPP=on

Hello,

I've been trying to build using my system's provided yaml-cpp (I tried both versions 0.5.3 and 0.6.1) but it consistently fails near the end of the build.

When I use the yaml-cpp provided as a submodule (under external_libraries/) and remove the -DSYSTEM_YAMLCPP=on CMake option, the build succeeds.

Here's the output:
```
[ 99%] Linking CXX static library libsclang.a
cd /tmp/guix-build-supercollider-3.9.1.drv-0/build/lang && /gnu/store/6g4pfzgmgwpfkwix0p7r6j349hffh2gs-cmake-3.7.2/bin/cmake -P CMakeFiles/libsclang.dir/cmake_clean_target.cmake
cd /tmp/guix-build-supercollider-3.9.1.drv-0/build/lang && /gnu/store/6g4pfzgmgwpfkwix0p7r6j349hffh2gs-cmake-3.7.2/bin/cmake -E cmake_link_script CMakeFiles/libsclang.dir/link.txt --verbose=1
/gnu/store/nnykzgwfy8mwh2gmxm715sjxykg8qjwn-binutils-2.28/bin/ar qc libsclang.a  CMakeFiles/libsclang.dir/LangPrimSource/SC_Wii.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrSignalPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrSched.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrPrimitive.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrMathPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/SC_ComPort.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/OSCData.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrArchiver.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrArrayPrimitives.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrBitPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrCharPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrFilePrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrListPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrPlatformPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrStringPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrSymbolPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrUnixPrim.cpp.o CMakeFiles/libsclang.dir/LangSource/AdvancingAllocPool.cpp.o CMakeFiles/libsclang.dir/LangSource/ByteCodeArray.cpp.o CMakeFiles/libsclang.dir/LangSource/DumpParseNode.cpp.o CMakeFiles/libsclang.dir/LangSource/GC.cpp.o CMakeFiles/libsclang.dir/LangSource/InitAlloc.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrInterpreter3.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrLexer.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrMathOps.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrMathSupport.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrMessage.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrObject.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrParseNode.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrSignal.cpp.o CMakeFiles/libsclang.dir/LangSource/PyrSymbolTable.cpp.o CMakeFiles/libsclang.dir/LangSource/SC_LanguageClient.cpp.o CMakeFiles/libsclang.dir/LangSource/SC_LanguageConfig.cpp.o CMakeFiles/libsclang.dir/LangSource/SC_TerminalClient.cpp.o CMakeFiles/libsclang.dir/LangSource/SimpleStack.cpp.o CMakeFiles/libsclang.dir/LangSource/VMGlobals.cpp.o CMakeFiles/libsclang.dir/LangSource/dumpByteCodes.cpp.o CMakeFiles/libsclang.dir/__/common/SC_Filesystem_macos.cpp.o CMakeFiles/libsclang.dir/__/common/SC_Filesystem_win.cpp.o CMakeFiles/libsclang.dir/__/common/SC_Filesystem_unix.cpp.o CMakeFiles/libsclang.dir/__/common/SC_Filesystem_iphone.cpp.o CMakeFiles/libsclang.dir/__/common/fftlib.c.o CMakeFiles/libsclang.dir/__/common/Samp.cpp.o CMakeFiles/libsclang.dir/__/common/SC_AllocPool.cpp.o CMakeFiles/libsclang.dir/__/common/SC_Reply.cpp.o CMakeFiles/libsclang.dir/__/common/SC_StringBuffer.cpp.o CMakeFiles/libsclang.dir/__/common/SC_StringParser.cpp.o CMakeFiles/libsclang.dir/__/common/SC_TextUtils.cpp.o CMakeFiles/libsclang.dir/__/editors/sc-ide/primitives/localsocket_utils.cpp.o CMakeFiles/libsclang.dir/__/common/sc_popen.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/PyrSerialPrim.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/SC_AlsaMIDI.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/SC_LID.cpp.o CMakeFiles/libsclang.dir/LangPrimSource/SC_HID_api.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/type_codec.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/metatype.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/debug.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/interface.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/factories.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/QcApplication.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/Common.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/QObjectProxy.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/QWidgetProxy.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/QcObjectFactory.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/QtDownload.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/hacks/hacks_x11.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/primitives.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_QObject.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_QPen.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_misc.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_QWidget.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_QPalette.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/primitives/prim_QImage.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/BasicWidgets.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcButton.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcListWidget.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcPopUpMenu.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcTreeWidget.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcScrollArea.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcAbstractStepValue.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcNumberBox.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcSlider2D.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcRangeSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcMultiSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcKnob.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcTextEdit.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcScope.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcScopeShm.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcLevelIndicator.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcWindow.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcGraph.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcCanvas.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/soundfileview/view.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/soundfileview/filestream.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/soundfileview/cachestream.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcFileDialog.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcPenPrinter.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/QcWebView.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/web_page.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/layouts/layouts.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/layouts/stack_layout.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/style/ProxyStyle.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/style/style.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/LanguageClient.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/moc_QcApplication.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/moc_QObjectProxy.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/moc_QWidgetProxy.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/moc_QtDownload.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_BasicWidgets.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcButton.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcCheckBox.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcTextField.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcListWidget.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcPopUpMenu.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcTreeWidget.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcScrollArea.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcNumberBox.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcSlider2D.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcRangeSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcMultiSlider.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcKnob.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcTextEdit.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcScope.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcScopeShm.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcGraph.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcLevelIndicator.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcCanvas.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/soundfileview/moc_view.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcFileDialog.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcPenPrinter.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcWebView.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_web_page.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/widgets/moc_QcWindow.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/layouts/moc_layouts.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/layouts/moc_stack_layout.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/style/moc_style.cpp.o CMakeFiles/libsclang.dir/__/QtCollider/moc_LanguageClient.cpp.o CMakeFiles/libsclang.dir/qrc_resources.cpp.o CMakeFiles/libsclang.dir/__/editors/sc-ide/primitives/sc_ipc_client.cpp.o CMakeFiles/libsclang.dir/__/editors/sc-ide/primitives/moc_sc_ipc_client.cpp.o CMakeFiles/libsclang.dir/__/SCDoc/SCDoc.cpp.o CMakeFiles/libsclang.dir/__/SCDoc/SCDoc.tab.cpp.o CMakeFiles/libsclang.dir/__/SCDoc/lex.scdoc.cpp.o CMakeFiles/libsclang.dir/__/SCDoc/SCDocPrim.cpp.o CMakeFiles/libsclang.dir/LangSource/Bison/lang11d_tab.cpp.o
/gnu/store/nnykzgwfy8mwh2gmxm715sjxykg8qjwn-binutils-2.28/bin/ranlib libsclang.a
make[2]: Leaving directory '/tmp/guix-build-supercollider-3.9.1.drv-0/build'
[ 99%] Built target libsclang
make[1]: Leaving directory '/tmp/guix-build-supercollider-3.9.1.drv-0/build'
make: *** [Makefile:164: all] Error 2
phase `build' failed after 1215.7 seconds
```

If my findings are not mistaken, it seems there is a link time dependency on yaml (on top of yaml-cpp). I re-run make inside the build directory, and got:

```
[ 69%] Linking CXX executable scide
cd /tmp/SuperCollider-Source/build/editors/sc-ide && /gnu/store/68l0b08gfbhfdfklp6zgvlh0v0hn1a1l-cmake-3.7.2/bin/cmake -E cmake_link_script CMakeFiles/SuperCollider.dir/link.txt --verbose=1
/gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/bin/c++    -msse -mfpmath=sse -msse2 -O2 -g -DNDEBUG   CMakeFiles/SuperCollider.dir/core/main.cpp.o CMakeFiles/SuperCollider.dir/core/sig_mux.cpp.o CMakeFiles/SuperCollider.dir/core/doc_manager.cpp.o CMakeFiles/SuperCollider.dir/core/sc_process.cpp.o CMakeFiles/SuperCollider.dir/core/sc_server.cpp.o CMakeFiles/SuperCollider.dir/core/sc_introspection.cpp.o CMakeFiles/SuperCollider.dir/core/sc_lexer.cpp.o CMakeFiles/SuperCollider.dir/core/settings/serialization.cpp.o CMakeFiles/SuperCollider.dir/core/settings/manager.cpp.o CMakeFiles/SuperCollider.dir/core/settings/theme.cpp.o CMakeFiles/SuperCollider.dir/core/session_manager.cpp.o CMakeFiles/SuperCollider.dir/core/util/standard_dirs.cpp.o CMakeFiles/SuperCollider.dir/core/util/scdoc_log.cpp.o CMakeFiles/SuperCollider.dir/widgets/cmd_line.cpp.o CMakeFiles/SuperCollider.dir/widgets/doc_list.cpp.o CMakeFiles/SuperCollider.dir/widgets/documents_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/editor_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/find_replace_tool.cpp.o CMakeFiles/SuperCollider.dir/widgets/lookup_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/main_window.cpp.o CMakeFiles/SuperCollider.dir/widgets/multi_editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/post_window.cpp.o CMakeFiles/SuperCollider.dir/widgets/session_switch_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/help_browser.cpp.o CMakeFiles/SuperCollider.dir/widgets/audio_status_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/lang_status_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/sc_editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/highlighter.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/line_indicator.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/overlay.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/autocompleter.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/completion_menu.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/general_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/sclang_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/editor_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/shortcuts_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/gui_utilities.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/text_format_list_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/popup_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/docklet.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/volume_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/status_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/style/style.cpp.o CMakeFiles/SuperCollider.dir/__/__/QtCollider/widgets/web_page.cpp.o CMakeFiles/SuperCollider.dir/__/__/common/SC_TextUtils.cpp.o CMakeFiles/SuperCollider.dir/__/__/common/SC_Filesystem_macos.cpp.o CMakeFiles/SuperCollider.dir/__/__/common/SC_Filesystem_win.cpp.o CMakeFiles/SuperCollider.dir/__/__/common/SC_Filesystem_unix.cpp.o CMakeFiles/SuperCollider.dir/__/__/common/SC_Filesystem_iphone.cpp.o CMakeFiles/SuperCollider.dir/primitives/localsocket_utils.cpp.o CMakeFiles/SuperCollider.dir/__/__/SCDoc/SCDoc.cpp.o CMakeFiles/SuperCollider.dir/__/__/SCDoc/SCDoc.tab.cpp.o CMakeFiles/SuperCollider.dir/__/__/SCDoc/lex.scdoc.cpp.o CMakeFiles/SuperCollider.dir/__/__/QtCollider/layouts/stack_layout.cpp.o CMakeFiles/SuperCollider.dir/core/moc_main.cpp.o CMakeFiles/SuperCollider.dir/core/moc_sig_mux.cpp.o CMakeFiles/SuperCollider.dir/core/moc_sc_process.cpp.o CMakeFiles/SuperCollider.dir/core/moc_sc_server.cpp.o CMakeFiles/SuperCollider.dir/core/moc_doc_manager.cpp.o CMakeFiles/SuperCollider.dir/core/settings/moc_manager.cpp.o CMakeFiles/SuperCollider.dir/core/moc_session_manager.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_cmd_line.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_doc_list.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_documents_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_editor_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_find_replace_tool.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_goto_line_tool.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_lookup_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_main_window.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_multi_editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_popup_text_input.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_post_window.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_session_switch_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_sessions_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_tool_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_help_browser.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_audio_status_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/moc_lang_status_box.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_sc_editor.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_highlighter.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_line_indicator.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_overlay.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_autocompleter.cpp.o CMakeFiles/SuperCollider.dir/widgets/code_editor/moc_completion_menu.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/moc_dialog.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/moc_general_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/moc_sclang_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/moc_editor_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/settings/moc_shortcuts_page.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_path_chooser_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_text_format_list_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_key_sequence_edit.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_popup_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_color_widget.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_docklet.cpp.o CMakeFiles/SuperCollider.dir/widgets/util/moc_volume_widget.cpp.o CMakeFiles/SuperCollider.dir/__/__/QtCollider/widgets/moc_web_page.cpp.o CMakeFiles/SuperCollider.dir/__/__/QtCollider/layouts/moc_stack_layout.cpp.o CMakeFiles/SuperCollider.dir/qrc_resources.cpp.o  -o scide -Wl,-rpath,:::::::::::::: -rdynamic /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Concurrent.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5WebKitWidgets.so.5.9.1 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5OpenGL.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5PrintSupport.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Sensors.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Quick.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Qml.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Sql.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Positioning.so.5.9.4 -lyaml ../../external_libraries/liboscpack.a -lboost_system -lboost_filesystem -lrt -lX11 -lpthread /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5WebKit.so.5.9.1 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Widgets.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Network.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Gui.so.5.9.4 /gnu/store/1mc7vildyyv40ib5kr2y07ay8pf8bcrm-profile/lib/libQt5Core.so.5.9.4 
ld: cannot find -lyaml
collect2: error: ld returned 1 exit status
make[2]: *** [editors/sc-ide/CMakeFiles/SuperCollider.dir/build.make:3001: editors/sc-ide/scide] Error 1
make[2]: Leaving directory '/tmp/SuperCollider-Source/build'
make[1]: *** [CMakeFiles/Makefile2:3600: editors/sc-ide/CMakeFiles/SuperCollider.dir/all] Error 2
make[1]: Leaving directory '/tmp/SuperCollider-Source/build'
make: *** [Makefile:164: all] Error 2
```
So it seems the SC-IDE has a dependency on "yaml" which I wasn't aware of, and which is somehow satisfied when using the bundled yaml-cpp. I'm not aware of another "yaml" library for C++ (else than yaml-cpp). Maybe the line 276 at https://github.com/supercollider/supercollider/blob/develop/editors/sc-ide/CMakeLists.txt should read yaml-cpp?

--- a/editors/sc-ide/CMakeLists.txt
+++ b/editors/sc-ide/CMakeLists.txt
@@ -199,6 +199,7 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common)
 include_directories(${CMAKE_SOURCE_DIR}/common)
 include_directories(${CMAKE_SOURCE_DIR}/include/plugin_interface)
 include_directories(${CMAKE_SOURCE_DIR}/external_libraries/boost)
+include_directories(${YAMLCPP_INCLUDE_DIR})
 # For QtCollider headers:
 include_directories(${CMAKE_SOURCE_DIR})
 # Needed for auto-generated forms headers:
@@ -273,7 +274,7 @@ endif()
 
 target_link_libraries( SuperCollider
     ${QT_IDE_LIBRARIES}
-    yaml
+    ${YAMLCPP_LIBRARY}
     oscpack
 )
 

--- a/external_libraries/CMakeLists.txt
+++ b/external_libraries/CMakeLists.txt
@@ -106,6 +106,8 @@ if(NOT YAMLCPP_FOUND)
   add_library(yaml STATIC EXCLUDE_FROM_ALL ${CMAKE_CURRENT_BINARY_DIR}/libyamlcpp.cpp)
   target_include_directories(yaml PUBLIC ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include boost)
   set_property( TARGET yaml PROPERTY FOLDER 3rdparty )
+  set(YAMLCPP_LIBRARY yaml)
+  set(YAMLCPP_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include)
 
   if(CMAKE_COMPILER_IS_GNUCXX)
     target_compile_options(yaml PRIVATE -Wno-deprecated-declarations)

--- a/lang/CMakeLists.txt
+++ b/lang/CMakeLists.txt
@@ -10,8 +10,6 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common
                     ${CMAKE_SOURCE_DIR}/include/server
                     ${CMAKE_SOURCE_DIR}/common
 
-                    ${YAMLCPP_INCLUDE_DIR}
-
                     ${CMAKE_SOURCE_DIR}/external_libraries/boost_sync/include
 
                     LangSource
@@ -19,6 +17,8 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common
 
                     ${CMAKE_SOURCE_DIR}/external_libraries/nova-tt
                     LangSource/Bison
+
+                    ${YAMLCPP_INCLUDE_DIR}
 )
 
 

